src/components/views/messages/EncryptionEvent.tsx

```diff
@@ -22,20 +22,27 @@ import { MatrixClientPeg } from '../../../MatrixClientPeg';
 import EventTileBubble from "./EventTileBubble";
 import MatrixClientContext from "../../../contexts/MatrixClientContext";
 import DMRoomMap from "../../../utils/DMRoomMap";
+import { objectHasDiff } from "../../../utils/objects";
 
 interface IProps {
     mxEvent: MatrixEvent;
 }
 
+const ALGORITHM = "m.megolm.v1.aes-sha2";
+
 const EncryptionEvent = forwardRef<HTMLDivElement, IProps>(({ mxEvent }, ref) => {
     const cli = useContext(MatrixClientContext);
     const roomId = mxEvent.getRoomId();
     const isRoomEncrypted = MatrixClientPeg.get().isRoomEncrypted(roomId);
 
-    if (mxEvent.getContent().algorithm === 'm.megolm.v1.aes-sha2' && isRoomEncrypted) {
+    if (!objectHasDiff(mxEvent.getPrevContent(), mxEvent.getContent())) return null; // nop
```

Comment:
`objectHasDiff` does a shallow comparison - is that okay..? if so, I think we should comment why that assumption holds

Reply:
It is okay because all MatrixEvents begin life as JSON so even identical objects are separate